crates/wit-bindgen/src/types.rs

```diff
@@ -40,6 +40,14 @@ impl std::ops::BitOrAssign for TypeInfo {
 
 impl Types {
     pub fn analyze(&mut self, resolve: &Resolve, world: WorldId) {
+        // Build up all type information first which is inherited through types,
+        // such as properties of borrows/lists/etc.
+        for (t, _) in resolve.types.iter() {
+            self.type_id_info(resolve, t);
+        }
```

Comment:
Does doing this step first mean that the calls to `self.type_info` in `type_info_func` should never traverse the whole type (due to the early exit if the entry exists already)?